build_tools/cmake/iree_filter_test.cmake

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
+
+function(iree_filter_test)
```

Comment:
Centralizing this logic is nice. Independent of the specifics of which labels are applied where, I think this is useful. If possible, it could help to sent an NFC PR with this and then adjusting which labels are applied in follow-ups.

I do wonder if we could make the usage less verbose... maybe using a macro and/or some standardized variable names?

```cmake
  # Current
  iree_filter_test(
    RESULT_VAR_ENABLED
      _ENABLED
    LABELS
      ${_RULE_LABELS}
    SIZE
      ${_RULE_SIZE}
    DRIVER
      ${_RULE_DRIVER}
  )
  if(NOT _ENABLED)
    return()
  endif()

  # Would this be nicer?
  iree_return_if_test_is_filtered()
```

Reply:
>  If possible, it could help to sent an NFC PR with this and then adjusting which labels are applied in follow-ups.

Yes, I'm slicing this into smaller PRs. The first is #14156.

> I do wonder if we could make the usage less verbose... maybe using a macro and/or some standardized variable names?

I was very tempted to do that. But I thought that hiding return statements would be evil.